source/common/buffer/watermark_buffer.cc

```diff
@@ -70,9 +70,16 @@ Api::IoCallUint64Result WatermarkBuffer::write(Network::IoHandle& io_handle) {
 
 void WatermarkBuffer::setWatermarks(uint32_t low_watermark, uint32_t high_watermark) {
   ASSERT(low_watermark < high_watermark || (high_watermark == 0 && low_watermark == 0));
+  if (overflow_watermark_multiplier_ > 0 &&
+      (overflow_watermark_multiplier_ * high_watermark) < high_watermark) {
+    ENVOY_LOG_MISC(error, "Error setting overflow threshold: overflow_waterflow_multiplier * "
```

Comment:
This will log spam if actually configured incorrectly. Can we downgrade this to debug or finally add a ENVOY_LOG_RATE_LIMIT type macro?

Reply:
Sure.